File: 2023-2024/sourcecode/mod3/lab10/SharedPreferencesBasicsJava/app/src/main/java/com/example/sharedpreferencesbasicsjava/viewmodel/UserViewModel.java
package com.example.sharedpreferencesbasicsjava.viewmodel;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;

import com.example.sharedpreferencesbasicsjava.repository.UserRepository;

public class UserViewModel extends AndroidViewModel {
    private final UserRepository repository = new UserRepository(getApplication());

    public UserViewModel(@NonNull Application application) {
        super(application);
    }

    public LiveData<String> getUsername() {
        return repository.getUsername();
    }

    public void clearUsers() {
        repository.clear();
    }

    public void addUser(String username) {
        repository.add(username);
    }
}

File: 2023-2024/sourcecode/mod2/lab9/SharedPreferencesBasicsKotlin/app/src/main/java/com/example/sharedpreferencesbasicskotlin/viewmodel/UserViewModel.kt
package com.example.sharedpreferencesbasicskotlin.viewmodel

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import com.example.sharedpreferencesbasicskotlin.repository.UserRepository
import kotlinx.coroutines.flow.MutableStateFlow
import kotlinx.coroutines.flow.StateFlow

class UserViewModel(application: Application) : AndroidViewModel(application) {
    private val repository: UserRepository = UserRepository(application)
    private val _username: MutableStateFlow<String> = MutableStateFlow(repository.username)

    val username: StateFlow<String>
        get() = _username

    fun addUsername(username: String) {
        repository.add(username)
        _username.value = username
    }

    fun clearUsername() {
        repository.clear()
        _username.value = ""
    }
}